clean-code-challanges/src/main/java/PigLatinTranslator.java

```diff
@@ -18,6 +21,46 @@
 public class PigLatinTranslator {
 
   public String translate(String englishPhrase) {
-    return null;
+    String[] words = splitAndReplace(englishPhrase);
+
+    for (int i = 0; i < words.length; i++) {
+      while (!beginsWithVowelSound(words[i]) && !beginsWithQu(words[i]) || firstLetterFollowedByY(words[i])) {
+        words[i] = moveNCharToEnd(words[i], 1);
+      }
+
+      if (beginsWithQu(words[i])) {
+        words[i] = moveNCharToEnd(words[i], 2);
+      }
+
+      words[i] = words[i] + "ay";
+    }
+
+    return String.join(" ", words);
+  }
+
+  private String[] splitAndReplace(String englishPhrase) {
+    String[] words = englishPhrase.split(" ");
+    for (int i = 0; i < words.length; i++) {
+      words[i] = words[i].replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
+    }
+    return words;
+  }
+
+  public boolean beginsWithVowelSound(String word) {
+    List<Character> vowelChars = Arrays.asList('a', 'i', 'e', 'o', 'u');
+    List<String> vowelStrings = Arrays.asList("xr", "yt");
+    return vowelChars.contains(word.charAt(0)) || vowelStrings.contains(word.substring(0, 2));
+  }
+
+  public boolean beginsWithQu(String word) {
+    return word.substring(0, 2).contains("qu");
+  }
+
+  public boolean firstLetterFollowedByY(String word) {
+    return word.substring(1,2).contains("y");
+  }
+
+  public String moveNCharToEnd(String word, int numberOfChar) {
```

Comment:
Du hast eine sehr smarte und kurze Lösung. Der Code ist jedoch keinesfalls zu kurz, denn der Code ist immernoch sehr gut zu lesen (vieleicht sogar gerade deshalb). Deine Methodennamen sind ebenfalls gut gewählt. Lediglich bei der moveNCharToEnd Methode bin ich mir nicht sicher, ob diese Mothode eine Anzahl an Zeichen oder nur den Buchstaben "N" nach hinten verschieben soll.

Reply:
Danke! Das ist ein guter Einwand, habe es mal in moveCharSetToEnd umgeschrieben.

Reply:
Top!